### PR TITLE
Adds functionality to create queries in workspaces

### PR DESCRIPTION
**Pull request by samuelechu**
image

### CODE REVIEW

File: src/main/webapp/components/query-editor/query-editor.tsx [hunk 1]
```diff
@@ -174,12 +162,7 @@ export default (props: EditorProps) => {
           marginTop: '10px',
         }}
       >
-        <Footer
```
Comment (samuelechu) [inline src/main/webapp/components/query-editor/query-editor.tsx]: Discussed with UX, and the plan is to remove manual saving here in favor of saving on search.

File: src/main/webapp/components/query-manager/add-query.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import SearchIcon from '@material-ui/icons/Search'
+import React from 'react'
+import { NavigationBarContext } from '../../nav-bar-context'
+import QueryEditorPopover from './query-editor-popover'
+import { AddQueryProps } from './types'
+const ReactDOM = require('react-dom')
+
+export default (props: AddQueryProps) => {
+  const [query, setQuery] = React.useState<{ id?: string }>({
+    id: new Date().toString(),
```
Comment (samuelechu) [inline src/main/webapp/components/query-manager/add-query.tsx]: creating queries with a mutation will be added in a later pr

File: src/main/webapp/components/workspace/workspace.js [hunk 3]
```diff
@@ -71,7 +71,7 @@ export default () => {
 
   const [listResults, setListResults] = React.useState([])
   const [currentQuery, setCurrentQuery] = useState(null)
-  const [queries, setQueries] = useState([])
+  const [queries, setQueries] = useState()
```
Comment (samuelechu) [inline src/main/webapp/components/workspace/workspace.js]: image
Having an initial state of undefined prevents this^ from showing before queries are fetched on a workspace that has existing queries.

File: src/main/webapp/components/query-manager/add-query.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  })
+  const { onSearch, QueryEditor } = props
+
+  const navBarLeftRef = React.useContext(NavigationBarContext)
```
Comment (willwill96) [inline src/main/webapp/components/query-manager/add-query.tsx]: :pencil2: make this variable name consistent across other components. In some places it's `navLeftRef`

**Pull request merged**